vpdq/cpp/io/vpdqio.cpp

```diff
@@ -176,6 +176,51 @@ bool readVideoDuration(
   durationInSec = (double)pFormatCtx->duration / MILLISEC_IN_SEC;
   return true;
 }
+
+bool readVideoFPS(
+    const string& inputVideoFileName,
+    double& framesPerSec,
+    const char* programName) {
+  AVFormatContext* pFormatCtx = avformat_alloc_context();
+  int rc =
+      avformat_open_input(&pFormatCtx, inputVideoFileName.c_str(), NULL, NULL);
+  if (rc != 0) {
+    fprintf(
+        stderr,
+        "%s: could not open video \"%s\".\n",
+        programName,
+        inputVideoFileName.c_str());
+    return false;
+  }
+  AVCodecContext* pCodecCtx;
+  int videoStream = -1;
+  rc = avformat_find_stream_info(pFormatCtx, NULL);
+  if (rc < 0) {
+    fprintf(
+        stderr,
+        "%s: could not find video stream info \"%s\".\n",
+        programName,
+        inputVideoFileName.c_str());
+    return false;
+  }
+  for (int i = 0; i < pFormatCtx->nb_streams; i++) {
+    if (pFormatCtx->streams[i]->codecpar->codec_type == AVMEDIA_TYPE_VIDEO &&
+        videoStream < 0) {
+      videoStream = i;
+    }
+  }
+  if (videoStream == -1) {
+    fprintf(
+        stderr,
+        "%s: could not find video stream \"%s\".\n",
+        programName,
+        inputVideoFileName.c_str());
+    return false;
+  }
+  AVRational fr = pFormatCtx->streams[videoStream]->avg_frame_rate;
```

Comment:
blocking: apart from these two lines this looks this is an exact copy of `readVideoResolution` imp. It seems like you could write a method that gets both at the same time. Is there a reason not to do that? 

(If you have to keep the utility of having methods for each for debugging you can have the existing methods call the new one and throw out the un-need result)

Reply:
Thanks for the suggestion. I was trying to separate the functionality, but I would combine them together.